src/Build.UnitTests/BackEnd/SdkResolution_Tests.cs

```diff
@@ -0,0 +1,155 @@
+﻿using System;
+using Microsoft.Build.BackEnd;
+using Microsoft.Build.BackEnd.Logging;
+using Microsoft.Build.Construction;
+using Microsoft.Build.Framework;
+using Microsoft.Build.UnitTests;
+using Microsoft.Build.UnitTests.BackEnd;
+using System.Collections.Generic;
+using System.Text;
+using Xunit;
+
+namespace Microsoft.Build.Engine.UnitTests.BackEnd
+{
+    public class SdkResolution_Tests
+    {
+        [Fact]
+        public void AssertFirstResolverCanResolve()
+        {
+            var log = new StringBuilder();
+            var sdk = new ReferencedSdk("1sdkName", "referencedVersion");
+            var logger = new MockLoggingService(message => log.AppendLine(message));
+            var bec = new BuildEventContext(0, 0, 0, 0, 0);
+            
+            SdkResolution resolution = new SdkResolution(new MockLoaderStrategy());
+            var result = resolution.GetSdkPath(sdk, logger, bec, new MockElementLocation("file"), "sln");
+
+            Assert.Equal("resolverpath1", result);
+            Assert.Equal("MockSdkResolver1 running", log.ToString().Trim());
+        }
+
+        [Fact]
+        public void AssertFirstResolverErrorsSupressedWhenResolved()
+        {
+            // 2sdkName will cause MockSdkResolver1 to fail with an error reason. The error will not 
+            // be logged because MockSdkResolver2 will succeed.
+            var log = new StringBuilder();
+            var sdk = new ReferencedSdk("2sdkName", "referencedVersion");
+            var logger = new MockLoggingService(message => log.AppendLine(message));
+            var bec = new BuildEventContext(0, 0, 0, 0, 0);
+
+            SdkResolution resolution = new SdkResolution(new MockLoaderStrategy());
+            var result = resolution.GetSdkPath(sdk, logger, bec, new MockElementLocation("file"), "sln");
+
+            var logResult = log.ToString();
+            Assert.Equal("resolverpath2", result);
+
+            // Both resolvers should run, and no ERROR string.
+            Assert.Contains("MockSdkResolver1 running", logResult);
+            Assert.Contains("MockSdkResolver2 running", logResult);
+
+            // Resolver2 gives a warning on success or failure.
+            Assert.Contains("WARNING2", logResult);
+            Assert.DoesNotContain("ERROR", logResult);
```

Comment:
Can wait, but: I don't think this is the right logging strategy. We should look at how NuGet handles multiple package sources and learn from their experience/mistakes. I think we should log every message from all resolvers but demote `ERROR` to `INFO`--otherwise, how will we diagnose the situation "the first resolver was supposed to resolve this but didn't"?